### DISCUSSION TITLE
Using functions as 'family' parameters correctly (operator==, hashCode)

### DISCUSSION THREAD

**Discussion by wujek-srujek** [rrousselGit/riverpod#651] [Q&A]
Here https://riverpod.dev/docs/concepts/modifiers/family#parameter-restrictions it says that parameters need to implement `operator==` and `hashCode` for 'family' to work. How would I do this correctly if I wanted a family which receives a callback as its parameter?
```dart
final provider = Provider.autoDispose.family<String, void Function()>(
  (state, callback) => ...,
);
```
I think inline lambdas or member methods, won't work, but global functions (or static ones) would - is this correct? I.e. I need to use the latter if I want 'family' to work correctly?

**Comment by rrousselGit** [comment]
You could try making sure that you're reusing an existing function instance.

So:

```dart
provider(() => ...)
```

is obviously a no-go.

But you could store that `() => ` somewhere and reuse it in different places


Although I'd suggest finding a different approach.
